src/lib/eina/eina_thread_win32.h

```diff
@@ -37,10 +37,28 @@
 #define EINA_THREAD_CANCEL_DEFERRED 0
 #define EINA_THREAD_CANCEL_ASYNCHRONOUS 1
 #define EINA_THREAD_CANCELED   ((void *) -1)
+
+struct _Eina_ThreadData
+{
+   /*
+    * In calling contains the argument to the thread func
+    * On exit, holds the return value of the thread
+    */
+   void *data;
+   void *(*func)(void *data);
+};
+
 /**
  * @typedef Eina_Thread
  * Type for a generic thread.
  */
-typedef HANDLE Eina_Thread;
+typedef struct _Eina_Thread
+{
+   HANDLE handle;
+   struct _Eina_ThreadData *data;
+} Eina_Thread;
 
-#endif 
\ No newline at end of file
+EINA_API DWORD eina_thread_self_id(void);
+EINA_API DWORD eina_thread_id(Eina_Thread t);
```

Comment:
I like the solution, altough I'm worried about having different APIs for Linux and Windows.

I'm talking to the community in `#edevelop` (IRC) to see it further if it would really be a problem.

Reply:
Hrm, IMHO, there is a design issue here, `eina_evlog` relies on `eina_thread_self` returning an integral type. In Posix systems, it returns a `pthread_t`, which according to Posix [is not guaranteed to be an integral type](https://www.man7.org/linux/man-pages/man3/pthread_self.3.html). My solution would be to introduce the `eina_thread_id` function in the official API and use it when the thread identifier needs to be an integral type, but I would like to get this merged and fixes that in a follow-up PR.